crates/aide/src/axum/routing.rs

```diff
@@ -381,3 +384,235 @@ method_router_top_level!(patch, patch_with);
+        /// See [`TypedPath`] for more details and examples.
+        fn typed_options<H, I, O, T, P>(self, handler: H) -> Self
+        where
+            H: axum::handler::Handler<T, S> + OperationHandler<I, O>,
+            T: SecondElementIs<P> + 'static,
+            I: OperationInput,
+            O: OperationOutput,
+            P: axum_extra::routing::TypedPath + schemars::JsonSchema + OperationInput;
+
+        /// Add a typed `PATCH` route to the router.
+        ///
+        /// The path will be inferred from the first argument to the handler function which must
+        /// implement [`TypedPath`].
+        ///
+        /// See [`TypedPath`] for more details and examples.
+        fn typed_patch<H, I, O, T, P>(self, handler: H) -> Self
+        where
+            H: axum::handler::Handler<T, S> + OperationHandler<I, O>,
+            T: SecondElementIs<P> + 'static,
+            I: OperationInput,
+            O: OperationOutput,
+            P: axum_extra::routing::TypedPath + schemars::JsonSchema + OperationInput;
+
+        /// Add a typed `POST` route to the router.
+        ///
+        /// The path will be inferred from the first argument to the handler function which must
+        /// implement [`TypedPath`].
+        ///
+        /// See [`TypedPath`] for more details and examples.
+        fn typed_post<H, I, O, T, P>(self, handler: H) -> Self
+        where
+            H: axum::handler::Handler<T, S> + OperationHandler<I, O>,
+            T: SecondElementIs<P> + 'static,
+            I: OperationInput,
+            O: OperationOutput,
+            P: axum_extra::routing::TypedPath + schemars::JsonSchema + OperationInput;
+
+        /// Add a typed `PUT` route to the router.
+        ///
+        /// The path will be inferred from the first argument to the handler function which must
+        /// implement [`TypedPath`].
+        ///
+        /// See [`TypedPath`] for more details and examples.
+        fn typed_put<H, I, O, T, P>(self, handler: H) -> Self
+        where
+            H: axum::handler::Handler<T, S> + OperationHandler<I, O>,
+            T: SecondElementIs<P> + 'static,
+            I: OperationInput,
+            O: OperationOutput,
+            P: axum_extra::routing::TypedPath + schemars::JsonSchema + OperationInput;
+
+        /// Add a typed `TRACE` route to the router.
+        ///
+        /// The path will be inferred from the first argument to the handler function which must
+        /// implement [`TypedPath`].
+        ///
+        /// See [`TypedPath`] for more details and examples.
+        fn typed_trace<H, I, O, T, P>(self, handler: H) -> Self
+        where
+            H: axum::handler::Handler<T, S> + OperationHandler<I, O>,
+            T: SecondElementIs<P> + 'static,
+            I: OperationInput,
+            O: OperationOutput,
+            P: axum_extra::routing::TypedPath + schemars::JsonSchema + OperationInput;
+    }
+
+    impl<S> ApiTypedPathExt<S> for crate::axum::ApiRouter<S>
```

Comment:
You don't need to use an extension trait for this.

```suggestion
    impl<S> crate::axum::ApiRouter<S>
```